styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -486,6 +490,23 @@ private void initializeResources(Storage storage, ShardedCounter shardedCounter,
     }
   }
 
+  @VisibleForTesting
+  void resetShards(final Storage storage, final Resource resource) {
```

Comment:
Maybe 'updateShards' and 'resetShards' should be part of 'Storage' interface. No need to refactor in this PR.